src/OpenLoco/src/GameCommands/ClearLand.cpp

```diff
@@ -73,36 +85,117 @@ namespace OpenLoco::GameCommands
+                    }
+                    removedBuildings.insert(buildingStart);
+
+                    World::TileManager::setRemoveElementPointerChecker(el);
+                    uint8_t removeBuildingFlags = flags;
+                    if ((flags & GameCommands::Flags::apply) || removedBuildings.size() != 1)
+                    {
+                        removeBuildingFlags |= GameCommands::Flags::flag_7;
+                    }
+                    if (flags & GameCommands::Flags::flag_6)
+                    {
+                        removeBuildingFlags &= ~(GameCommands::Flags::flag_6 | GameCommands::Flags::apply);
+                    }
+                    GameCommands::BuildingRemovalArgs args{};
+                    args.pos = buildingStart;
+                    Interop::registers regs = static_cast<Interop::registers>(args);
+                    regs.bl = removeBuildingFlags;
+                    removeBuilding(regs);
+                    if (regs.ebx == GameCommands::FAILURE)
+                    {
+                        return TileManager::ClearFuncResult::collisionErrorSet;
+                    }
+                    if (flags & GameCommands::Flags::apply)
+                    {
+                        S5::getOptions().madeAnyChanges = 1;
+                    }
+                    cost += regs.ebx;
+
+                    if (!(flags & GameCommands::Flags::apply) || flags & GameCommands::Flags::flag_6)
+                    {
+                        return TileManager::ClearFuncResult::noCollision;
+                    }
+                    if (World::TileManager::wasRemoveOnLastElement())
+                    {
+                        return TileManager::ClearFuncResult::allCollisionsRemoved;
+                    }
+                    return TileManager::ClearFuncResult::collisionRemoved;
+                }
+                default:
+                    return TileManager::ClearFuncResult::noCollision;
+            }
+        };
 
-        // TODO: implement 0x00469E07 as a real function after canConstructAt is implemented
         auto tileHeight = World::TileManager::getHeight(pos);
-        if (TileManager::sub_462908(pos, tileHeight.landHeight / 4, tileHeight.landHeight / 4, qt, 0x00469E07))
-            return _sub469E07Cost;
+        if (TileManager::sub_462908(pos, tileHeight.landHeight / 4, tileHeight.landHeight / 4, qt, clearFunc))
+            return cost;
         else
             return GameCommands::FAILURE;
     }
 
     // 0x00469CCB
     static uint32_t clearLand(const ClearLandArgs& args, const uint8_t flags)
     {
-        _F00170 = 0xFFFF;
         _F0013C = args.pointB.x << 16 | args.pointA.x;
         _F00140 = args.pointB.y << 16 | args.pointA.y;
         _F00144 = flags;
 
         World::TilePosRangeView tileLoop{ { args.pointA }, { args.pointB } };
         uint32_t totalCost = 0;
+        std::set<World::Pos3, LessThanPos3> removedBuildings{};
```

Comment:
Perhaps add a comment here detailing that this set is used across `clearTile` calls.